### PR TITLE
Change error-handling across handlers. 

### PR DESCRIPTION
**Pull request by Mathew-Estafanous**
issue #11393 

This is a continuation of a previous PR (#11599). Transition all the remaining endpoints to returning an error as opposed to writing directly to the response body.

### CODE REVIEW

File: agent/txn_endpoint.go [hunk 1]
```diff
@@ -87,13 +87,10 @@ func (s *HTTPHandlers) convertOps(resp http.ResponseWriter, req *http.Request) (
...
 	if req.ContentLength > maxTxnLen {
-		resp.WriteHeader(http.StatusRequestEntityTooLarge)
-		fmt.Fprintf(resp,
-			"Request body(%d bytes) too large, max size: %d bytes. See %s.",
-			req.ContentLength, maxTxnLen,
-			"https://www.consul.io/docs/agent/options.html#txn_max_req_len",
-		)
-		return nil, 0, false
+		return nil, 0, EntityTooLargeError{
+			Reason: fmt.Sprintf("Request body too large, max size: %d bytes. See %s.",
```
Comment (kisunji) [inline agent/txn_endpoint.go:91]: This is missing `(%d bytes)` from the original

**Review by kisunji — CHANGES_REQUESTED** [review]
Thanks again for your contribution! One minor comment about error string and could you remove `agent/uiserver/bindata_assetfs.go` from the PR (we try to generate it by CI)?

**Comment by Mathew-Estafanous** [comment]
> could you remove `agent/uiserver/bindata_assetfs.go` from the PR (we try to generate it by CI)?

I’m not exactly sure how I could undo that change since I already pushed it. I believe the change happened when I used ``gofmt -s -w``.

Is there a way to go back to the commit history and unstage that specific file from the commit history? I look up how to do that, but I’m worried about messing up the git history.
Would using this work? ``git filter-branch --force --index-filter 'git rm --cached --ignore-unmatch ./agent/uiserver/bindata_assetfs.go' --prune-empty --tag-name-filter cat -- --all``

**Comment by kisunji** [comment]
> > could you remove `agent/uiserver/bindata_assetfs.go` from the PR (we try to generate it by CI)?
> 
> I’m not exactly sure how I could undo that change since I already pushed it. I believe the change happened when I used `gofmt -s -w`.
> 
> Is there a way to go back to the commit history and remove that specific file from the commit history? I looked up how to do that, but I’m worried about messing up the git history. Would using this work? `git filter-branch --force --index-filter 'git rm --cached --ignore-unmatch ./agent/uiserver/bindata_assetfs.go' --prune-empty --tag-name-filter cat -- --all`

Since this is your own branch, you can simply `git checkout main -- agent/uiserver/bindata_assetfs.go` and push a new commit. I'll be squashing your commits into one so there isn't really a need to keep your history clean

**Comment by kisunji** [comment]
Out of scope for this PR but might be interesting for you to think about:

We have a bunch of custom error types now for common HTTP Status Codes - most of them have a `Reason string` field and a custom type assertion function in `http.go`. One downside of our approach is that any new status code we want to return requires the creation of a new error type and assertion func (as you've done with `EntityTooLargeError` in this PR)

Do you think we could refactor these into a more general HTTP error type?

**Review by kisunji — APPROVED** [review]

**Comment by hc-github-team-consul-core** [comment]
:cherries: If backport labels were added before merging, cherry-picking will start automatically.

To retroactively trigger a backport after merging, add backport labels and re-run https://circleci.com/gh/hashicorp/consul/570828.

**Pull request merged**